File: benchmarks/java/B2BubbleSort.java
import java.util.Random;
public class B2BubbleSort {
    public static void main(String[] args) {
        int N = 100000;
        Random rng = new Random(42);
        int[] arr = new int[N];
        for (int i = 0; i < N; i++) arr[i] = rng.nextInt(N);
        for (int i = 0; i < N - 1; i++) {
            boolean swapped = false;
            for (int j = 0; j < N - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int tmp = arr[j]; arr[j] = arr[j + 1]; arr[j + 1] = tmp;
                    swapped = true;
                }
            }
            if (!swapped) break;
        }
        System.out.println(arr[0] + " " + arr[N - 1]);
    }
}

File: benchmarks/kotlin/b2_bubble_sort.kt
import kotlin.random.Random
fun main() {
    val N = 100_000
    val rng = Random(42)
    val arr = IntArray(N) { rng.nextInt(N) }
    for (i in 0 until N - 1) {
        var swapped = false
        for (j in 0 until N - i - 1) {
            if (arr[j] > arr[j + 1]) { val t = arr[j]; arr[j] = arr[j+1]; arr[j+1] = t; swapped = true }
        }
        if (!swapped) break
    }
    println("${arr[0]} ${arr[N-1]}")
}
